enterprise/frontend/src/metabase-enterprise/google_drive/GdriveAddDataPanel.tsx

```diff
@@ -173,14 +221,19 @@ export const GdriveAddDataPanel = () => {
   return (
     <PanelWrapper
       isModalOpen={isConnectionModalOpen}
-      onModalClose={closeConnectionModal}
+      // We are closing the parent (Add data) modal so we don't have to explicitly
+      // close the connection modal as well. It will be removed from DOM with its parent gone.
+      onModalClose={onAddDataModalClose}
```

Comment:
This means that it closes both when a user initiated an action (click on X, press ESC) **and** when it closes alone (upon successful connection mutation).

I couldn't find an easy and elegant way to separate the two without introducing more props.

Reply:
https://github.com/user-attachments/assets/75d77a06-cf21-4d02-86e8-674516d5b9f7